### PR TITLE
add ordered merge proc

### PR DESCRIPTION
**Pull request by alfred-landrum**
I'm splitting this out of some ongoing work for https://github.com/brimsec/zq/pull/1074 . This is also the Proc part of the work for https://github.com/brimsec/zq/issues/1036 .

### CODE REVIEW

File: proc/orderedmerge.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+				if res.Batch == nil {
+					m.done[i] = true
+					continue
+				}
+				m.batches[i] = res.Batch
+			case <-m.Context.Done():
+				return nil, m.Context.Err()
+			}
+		}
+		if idx == -1 || m.compare(m.batches[i], m.batches[idx]) {
```
Comment (henridf) [inline proc/orderedmerge.go:79]: Unless I'm missing something, I don't think we can always use batch-by-batch comparison here. I think we need to do record by record (like `zbuf.Combiner` does, and like sort/groupby do when reading from spills). 

(Batch-by-batch could be an optimization for cases where we know that batches are non-overlapping in the ordering key, but I'd suggest we add that at a later stage if necessary).

Comment (alfred-landrum) [reply proc/orderedmerge.go:79]: Thanks, I'll adjust. For the use case with archives, I was working on the assumption that there is no overlap between the batches we use for the parallel scanner, but that's a very narrow use case that doesn't work well in this general proc.

Comment (alfred-landrum) [reply proc/orderedmerge.go]: I've added the per-record changes & a test, PTAL!

File: proc/orderedmerge.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+)
+
+// An OrderedMerge proc merges multiple upstream inputs into one output.
+// If the input streams are ordered according to the configured comparison,
+// the output of OrderedMerge will have the same order.
+type OrderedMerge struct {
+	Base
+	recCmp  zbuf.RecordCmpFn
```
Comment (henridf) [inline proc/orderedmerge.go:15]: This seems like it should be `proc.CompareFn`

File: proc/orderedmerge_test.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		for _, r := range b.Records() {
+			if err := w.Write(r); err != nil {
+				return "", err
+			}
+		}
+		b.Unref()
+	}
+}
+
+func fieldReadCompare(field string, reversed bool) (zbuf.RecordCmpFn, error) {
```
Comment (henridf) [inline proc/orderedmerge_test.go:69]: nit: `reversed` isn't used in this func

Comment (henridf) [reply proc/orderedmerge_test.go:69]: (also we could use expr.NewCompareFn instead of this).

File: proc/orderedmerge_test.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zio/tzngio"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var omTestInputs = []string{
```
Comment (henridf) [inline proc/orderedmerge_test.go:22]: nit: place this closer to the test outputs (`cases`)

**Pull request merged**